File: TripPartner.WebAPI/Domain_Models/Trip.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;
using TripPartner.WebAPI.Models;

namespace TripPartner.WebAPI.Domain_Models
{
    public class Trip
    {
        [Key]
        public int Id { get; set; }

        [DisplayName("Trip started")]
        public DateTime DateStarted { get; set; }

        [DisplayName("Trip ended")]
        public DateTime DateEnded { get; set; }
        public string CreatorId { get; set; }
        [ForeignKey("CreatorId")]
        public ApplicationUser Creator { get; set; }
        public int DestinationId { get; set; }
         [ForeignKey("DestinationId")]
         public Location Destination { get; set; }
         public int OriginId { get; set; }
         [ForeignKey("OriginId")]
         public Location Origin { get; set; }
        public virtual ICollection<Story> Stories { get; set; }

    }
}

File: TripPartner.Web/app/components/trips/services/Trip.js
﻿(function (module) {

    module.factory('Trip', ['Location', function (Location) {
        function Trip(model) {
            this.Id = model.Id,
            this.DateStarted = model.DateStarted,
            this.DateEnded = model.DateEnded,
            this.Destination = new Location(model.Destination),
            this.Origin = new Location(model.Origin),
            this.CreatorId = model.CreatorId,
            this.CreatorUsername = model.CreatorUsername
        }
        return Trip;
    }]);

})(angular.module('app.trips'));